File: page-objects/java-fluentlenium/complete/src/main/java/pages/SearchResultsPage.java
package pages;

import org.fluentlenium.core.FluentPage;
import org.fluentlenium.core.annotation.Page;
import org.openqa.selenium.By;

import java.util.concurrent.TimeUnit;

@SuppressWarnings("unused")
public class SearchResultsPage extends FluentPage {

    @Page
    private ProductDetailPage productDetailPage;

    public ProductDetailPage clickOnFirstProduct() {
        find(By.cssSelector("z-grid[class='z-nvg-cognac_articles'] > z-grid-item:first-child")).click();
        return productDetailPage;
    }
}

File: page-objects/js-webdriverio/complete/pageobjects/searchResults.page.js
// searchResults.page.js
var Page = require('./page');
var ProductDetailPage = require('./productDetail.page');

var SearchResultsPage = Object.create(Page, {
    /**
     * define elements
     */
    firstElement: { get: function () { return browser.element("z-grid[class='z-nvg-cognac_articles'] > z-grid-item:first-child"); } },

    clickOnFirstElement: { value: function() {
            this.firstElement.click();
            return ProductDetailPage;
        } }
});

module.exports = SearchResultsPage;